build_tools/azure/update_environments_and_lock_files.py

```diff
@@ -0,0 +1,406 @@
+"""Script to update CI environment files and associated lock files.
+
+To run it you need to be in the root folder of the scikit-learn repo:
+python build_tools/azure/update_environments_and_lock_files.py
+
+Two scenarios where this script can be useful:
+- make sure that the latest versions of all the dependencies are used in the CI.
+  We can run this script regularly and open a PR with the changes to the lock
+  files. This workflow will eventually be automated with a bot in the future.
+- bump minimum dependencies in sklearn/_min_dependencies.py. Running this
+  script will update both the CI environment files and associated lock files.
+  You can then open a PR with the changes.
+
+Environments are conda environment.yml or pip requirements.txt. Lock files are
+conda-lock lock files or pip-compile requirements.txt.
+
+pip requirements.txt are used when we install some dependencies (e.g. numpy and
+scipy) with apt-get and the rest of the dependencies (e.g. pytest and joblib)
+with pip.
+
+"""
+
+import re
+import subprocess
+import sys
+from pathlib import Path
+import shlex
+import json
+
+from jinja2 import Environment
+
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+logger.addHandler(handler)
+
+
+common_dependencies_without_coverage = [
+    "python",
+    "numpy",
+    "blas",
+    "scipy",
+    "cython",
+    "joblib",
+    "threadpoolctl",
+    "matplotlib",
+    "pandas",
+    "pyamg",
+    "pytest",
+    "pytest-xdist",
+    "pillow",
+]
+
+common_dependencies = common_dependencies_without_coverage + [
+    "codecov",
+    "pytest-cov",
+    "coverage",
+]
+
+docstring_test_dependencies = ["sphinx", "numpydoc"]
+
+default_package_constraints = {
+    # XXX: pytest is temporary pinned to 6.2.5 because pytest 7 causes CI
+    # issues https://github.com/scikit-learn/scikit-learn/pull/22381
+    "pytest": "6.2.5",
```

Comment:
For my understanding: If we were to unpin `pytest` in the future, one would need to run `build_tools/azure/update_environments_and_lock_files.py` to regenerate the lock files again. Is that correct?

Reply:
Yes exactly!